scene/gui/link_button.cpp

```diff
@@ -139,6 +149,20 @@ Size2 LinkButton::get_minimum_size() const {
 	return text_buf->get_size();
 }
 
+void LinkButton::open_url() {
+	ERR_FAIL_COND_MSG(url == "", "Url is empty.");
+
+	_open_url();
+}
+
+void LinkButton::_open_url() {
+	if (url == "") {
+		return;
+	}
+
+	OS::get_singleton()->shell_open(url);
+}
```

Comment:
Is there any strong reason to have `open_url()` when users can just do `OS.shell_open(mylinkbutton.get_url())`? I think the most common case here is when users click on the button (the private method), so we really only need one method here.

We should use the `.is_empty()` helper method on String because it's faster and more readable.

```suggestion
void LinkButton::_open_url() {
	if (url.is_empty()) {
		return;
	}

	OS::get_singleton()->shell_open(url);
}
```

Reply:
> Is there any strong reason to have open_url() when users can just do OS.shell_open(mylinkbutton.get_url())? I think the most common case here is when users click on the button (the private method), so we really only need one method here.

You are right. The only use for `open_url()` would be when you want to connect something to it. But you usually connect buttons, not to buttons, so this use-case is not really valid.